### DISCUSSION TITLE
Standardize feature request handling

### DISCUSSION THREAD

**Discussion by frenck** [home-assistant/architecture#497] [General] [answered]
## Context

We currently have inconsistent handling of our feature requests across our organization.

We have our community forum: <https://community.home-assistant.io/c/feature-requests/13>
But, in some places (like the [frontend repository](https://github.com/home-assistant/frontend/issues?q=is:open+is:issue+label:feature-request). We keep issues labeled with `feature-request`.

On other repo's we close feature request issues and redirect people actively to the forum (like on the core repository).

## Proposal

The proposal is as follows:

>  Enable GitHub Discussions on our main repositories to handle Feature Requests.

Recently GitHub brought in the Discussions feature, you are looking at now.
And for the frontend project, we have actually been playing with the thing:

See here: <https://github.com/home-assistant/frontend/discussions>

![image](https://user-images.githubusercontent.com/195327/106397491-06b86680-640e-11eb-9c00-f66c7c23c0ca.png)

While the frontend still has to move a lot of issues to discussions, I think this feature combined with the categories might actually work for things like the core too.

I've been part of the private feedback group for Discussions @ GitHub as well, and one of the things we requested and was added (based on feature requests in mind) is the ability to vote on topics made.

![image](https://user-images.githubusercontent.com/195327/106397576-7fb7be00-640e-11eb-8d3d-53cb3153ad9c.png)\

Lastly, very recently, GitHub added the possibility to filter notifications:

![image](https://user-images.githubusercontent.com/195327/106397355-3fa40b80-640d-11eb-8568-04d4534aff18.png)

Allowing one to unsubscribe from discussion (and thus feature request) notifications. This was the last part I considered a requirement before being able to propose this. 

Hence 👋 

~~I know that Pascal would like this on the Supervisor repository and Stefan once said he was interested in the Operating System repository as well.~~ <- Edit: At least for Pascal this was incorrect from my memory. See his comment below.

 However, being consistent across the org would be nice. Hence this proposal.

## Consequences

- The feature request becomes more connected with the environment where they are implemented (on GitHub / the codebase).
- Categorization of feature requests, fitting for the repo they belong to (for example, the frontend has a "Lovelace" category)
- This might raise the bar a bit for people that are scared of GitHub, however, this barrier is lowered for this constantly.
- Issues with feature requests, can be moved to discussions (instead of brutally close them).
- Standardized with other repositories. One method to handle those across the organization.
- Issues, PRs & Discussions can be linked up and referenced together.

## Conclusion

Not standardizable across the org, which was the sole purpose of this proposal. Not implementing.
See: https://github.com/home-assistant/architecture/discussions/497#discussioncomment-325843

**Comment by balloob** [comment]
Makes sense but since most will disable notifications, discussions for any other purpose will be unusable.

**Comment by Kane610** [comment]
I like it, I haven't looked many times in the feature request part of the forums. This might help keep them a bit more available and viewable, also references will work better. If labels for integrations worked here it could also tag maintainers as well

**Reply by frenck** [reply]
Labels are much requested in the feedback of the Discussions features, unfortunately, they don't exist.

**Comment by pvizeli** [comment]
> I know that Pascal would like this on the Supervisor

Of curse not. I hate feature requests. I would like to have that on the community forum, but I can live with the core repo. Please don't enable this feature request on the supervisor.

**Reply by frenck** [reply]
OK in that case I remembered that incorrectly, sorry m8. Noted.

**Comment by frenck** [comment, marked as answer]
> Of curse not. I hate feature requests. I would like to have that on the community forum, but I can live with the core repo. Please don't enable this feature request on the supervisor.

I guess that renders this proposal useless.

**Conclusion: Not implementing.**

**Comment by raman325** [comment]
I know this discussion is already dead but in case it gets reopened in the future, I support this. Similar to what @Kane610 said I have a hard time browsing the feature requests area of the community forums. Having labels is great, and I think with some careful thought, we could leverage the discussions feature well.

One way would be to create a feature request discussion for every integration and let people add top level comments for their feature requests which can get up and down voted. Once a feature is implemented the thread can get removed from the discussion.

I'd also be open to alternative ways of handling it, just throwing out an idea for a potential future discussion
